libicyque.c

```diff
@@ -1618,36 +1621,24 @@ icq_session_start(IcyQueAccount *ia)
 {
 	if (!ICYQUE_ACCOUNT_IS_VALID(ia)) return;
 	
-	const gchar *url = ICQ_API_SERVER "/aim/startSession";
-	GString *postdata = g_string_new(NULL);
-	
-	// Make sure these are added alphabetically for the signature to work
-	g_string_append_printf(postdata, "a=%s&", purple_url_encode(ia->token));
-	// g_string_append_printf(postdata, "assertCaps=%s&", purple_url_encode(ICQ_ASSERT_CAPS));
-	// g_string_append_printf(postdata, "buildNumber=%s&", purple_url_encode("23341"));
-	// g_string_append_printf(postdata, "clientName=%s&", purple_url_encode("ICQ"));
-	g_string_append_printf(postdata, "deviceId=%s&", purple_url_encode(ia->device_id));
-	g_string_append_printf(postdata, "events=%s&", purple_url_encode(ICQ_EVENTS));
-	g_string_append(postdata, "f=json&");
-	g_string_append(postdata, "imf=plain&");
-	g_string_append_printf(postdata, "includePresenceFields=%s&", purple_url_encode(ICQ_PRESENCE_FIELDS));
-	// g_string_append_printf(postdata, "interestCaps=%s&", purple_url_encode("094613504C7F11D18222444553540000,094613514C7F11D18222444553540000,8EEC67CE70D041009409A7C1602A5C84"));
-	g_string_append(postdata, "invisible=false&");
-	g_string_append_printf(postdata, "k=%s&", purple_url_encode(ICQ_DEVID));
-	g_string_append(postdata, "language=en-US&");
-	g_string_append(postdata, "rawMsg=0&");
-	g_string_append(postdata, "sessionTimeout=31536000&");
-	//g_string_append(postdata, "sig_sha256_force=1&");
-	g_string_append_printf(postdata, "ts=%d&", (int)(time(NULL) - ia->server_time_offset));
-	g_string_append(postdata, "view=online"); //todo mobile?
-	
-	gchar *sig_sha256 = icq_get_url_sign(ia, TRUE, url, postdata->str);
-	g_string_append_printf(postdata, "&sig_sha256=%s", purple_url_encode(sig_sha256));
-	g_free(sig_sha256);
-	
-	icq_fetch_url_with_method(ia, "POST", url, postdata->str, icq_session_start_cb, NULL);
-	
-	g_string_free(postdata, TRUE);
+	GString *url = g_string_new(ICQ_API14_SERVER "/aim/startSession?");
+
+	g_string_append_printf(url, "a=%s&", purple_url_encode(ia->token));
+	g_string_append_printf(url, "ts=%d&", (int)(time(NULL) - ia->server_time_offset));
+	g_string_append_printf(url, "k=%s&", purple_url_encode(ICQ_DEVID));
+	g_string_append(url, "view=online&");
+	g_string_append(url, "clientName=webicq&");
+	g_string_append(url, "language=en-US&");
+	g_string_append_printf(url, "deviceId=%s&", purple_url_encode(ia->device_id));
+	g_string_append(url, "sessionTimeout=31536000&");
+	g_string_append_printf(url, "assertCaps=%s&", purple_url_encode(ICQ_ASSERT_CAPS));
+	g_string_append(url, "interestCaps=&");
+	g_string_append_printf(url, "events=%s&", purple_url_encode(ICQ_EVENTS));
+	g_string_append_printf(url, "includePresenceFields=%s", purple_url_encode(ICQ_PRESENCE_FIELDS));

```

Comment:
This call doesn't need the sig_sha256 hash?

Reply:
Not anymore, or at least, not on this API endpoint, which has been taken from the Web app.